File: kora-java-crud-submodule/kora-java-crud-submodule-pet-api/src/test/java/ru/tinkoff/kora/example/submodule/pet/TestPetApplication.java
package ru.tinkoff.kora.example.submodule.pet;

import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;
import ru.tinkoff.kora.config.hocon.HoconConfigModule;
import ru.tinkoff.kora.example.submodule.pet.service.PetService;

@KoraApp
public interface TestPetApplication extends
        HoconConfigModule,
        PetModule {

    @Root
    default String root(PetService petService) {
        return "root";
    }
}

File: examples/kotlin/kora-kotlin-crud-submodule/kora-kotlin-crud-submodule-pet-api/src/test/kotlin/ru/tinkoff/kora/kotlin/example/submodule/pet/TestPetApplication.kt
package ru.tinkoff.kora.kotlin.example.submodule.pet

import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.config.hocon.HoconConfigModule
import ru.tinkoff.kora.kotlin.example.submodule.pet.service.PetService

@KoraApp
interface TestPetApplication : HoconConfigModule, PetModule {

    @Root
    fun root(petService: PetService): String = "root"
}
